Automattic/wp-calypso
# FSE: Toggle editor post title visibility with a filter / Set block template editor setting

### Changes proposed in this Pull Request

#### Toggle editor post title visibility with a filter

This PRs implements a more robust approach than https://github.com/Automattic/wp-calypso/pull/34214 as suggested by @gwwar in order to avoid potential rendering flashes (https://github.com/Automattic/wp-calypso/pull/34214#discussion_r297425287). 

Now, the `a8c/post-content` block always renders a post title before the content, but is hidden by default. It will show up when the `admin_body_class` filter adds the `show-post-title-before-content` class, which will also hide the regular editor post title. This makes unnecessary the selector `isFullSitePage` added in https://github.com/Automattic/wp-calypso/pull/34214, so it has been removed on this PR.

#### Set block template editor setting

The above caused a side effect. Since the FSE plugin creates a default template that is assigned to all pages without a specific template, we would get the `show-post-title-before-content` class when creating a new page. This was making the post title to be missing because at the moment we are not applying the template parts when we start a new post.

This has been fixed on this PR by defining a block template on the `template` editor setting

### Follow-up

This PR will introduce a couple of known issues:
- Seems that the page templates (_Portfolio_, _Services_, etc) assume the page will be empty. So now that we apply the FSE template when starting a new page, selecting any other template than _Blank_ will result in the content of the selected page template to be appended after the footer template part.
- The `a8c/post-content` block can be removed by the user, resulting in a missing post title since the `show-post-title-before-content` would keep the regular editor post title hidden.

Should we fix them before landing this PR?

### Testing instructions

- Apply this branch to your FSE testing environment.
- Trash all templates and template parts.
- Deactivate and then reactivate the FSE plugin.
- Create a new page. 
- Select the _Blank_ template. 
- Make sure the default FSE template is applied.
- Make sure the title of the page is displayed below the header template part and at the beginning of the post content block.
- Try inserting a post content blog a not full site page (i.e. in a regular post or in template).
- Verify the post title is visible in the editor and the default FSE template is not applied.
